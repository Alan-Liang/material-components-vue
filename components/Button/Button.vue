<template>
    <button class="mdc-button" :class="classes" v-bind="$attrs" v-on="$listeners">
        <a v-if="icon" class="material-icons mdc-button__icon">{{icon}}</a>
        <slot />
    </button>
</template>

<script>
import { MDCRipple } from '@material/ripple'
export default {
    props: {
        icon: {
            type: String,
            required: false
        },
        raised: {
            type: Boolean,
            required: false
        },
        unelevated: {
            type: Boolean,
            required: false
        },
        stroked: {
            type: Boolean,
            required: false
        },
        dense: {
            type: Boolean,
            required: false
        },
        compact: {
            type: Boolean,
            required: false
        },
        interactive: {
            type: Boolean,
            required: false
        }
    },
    data() {
        return {
            mdcRipple: null
        }
    },
    computed: {
        classes() {
            return {
                'mdc-button--raised': this.raised,
                'mdc-button--unelevated': this.unelevated,
                'mdc-button--stroked': this.stroked,
                'mdc-button--dense': this.dense,
                'mdc-button--compact': this.compact
            }
        }
    },
    mounted() {
        if (this.interactive)
            this.mdcRipple = new MDCRipple(this.$el)
    },
    destroyed() {
        if (this.interactive)
            this.mdcRipple.destroy()
    }
}
</script>

