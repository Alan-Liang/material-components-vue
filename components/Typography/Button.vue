<template>
  <span class="mdc-typography--button" :class="classes">
    <slot />
  </span>
</template>

<script>
export default {
  props: {
    adjustMargin: {
      type: Boolean,
      default: false
    }
  },
  computed: {
    classes () {
      return {
        'mdc-typography--adjust-margin': this.adjustMargin
      }
    }
  }
}
</script>

<style lang="scss">
@import "@material/typography/mdc-typography";
</style>
