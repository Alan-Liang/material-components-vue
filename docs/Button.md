# Button

## Text

<ComponentLayout>
<ComponentWrapper>
<m-button>baseline</m-button>
</ComponentWrapper>
<ComponentWrapper>
<m-button interactive>interactive</m-button>
</ComponentWrapper>
<ComponentWrapper>
<m-button>
<m-icon slot="icon" icon="cloud"/>
icon</m-button>
</ComponentWrapper>
<ComponentWrapper>
<m-button dense>dense</m-button>
</ComponentWrapper>
<ComponentWrapper>
<m-button disabled>disabled</m-button>
</ComponentWrapper>
</ComponentLayout>

