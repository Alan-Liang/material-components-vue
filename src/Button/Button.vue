<template>
    <button class="mdc-button" :class="classes" :id="id" v-bind="$attrs" v-on="$listeners">
        <a v-if="icon" class="material-icons mdc-button__icon">{{icon}}</a>
        <slot>
            {{text}}
        </slot>
    </button>
</template>

<script>
import { MDCRipple } from '@material/ripple'
export default {
    props: {
        text: {
            type: String,
            required: false
        },
        id: {
            type: String,
            required: false
        },
        icon: {
            type: String,
            required: false
        },
        raised: {
            type: Boolean,
            required: false
        },
        unelevated: {
            type: Boolean,
            required: false
        },
        stroked: {
            type: Boolean,
            required: false
        },
        dense: {
            type: Boolean,
            required: false
        },
        compact: {
            type: Boolean,
            required: false
        },
        interactive: {
            type: Boolean,
            required: false
        }
    },
    data() {
        return {
            mdcRipple: null
        }
    },
    computed: {
        classes() {
            return {
                'mdc-button--raised': this.raised,
                'mdc-button--unelevated': this.unelevated,
                'mdc-button--stroked': this.stroked,
                'mdc-button--dense': this.dense,
                'mdc-button--compact': this.compact
            }
        }
    },
    mounted() {
        if (this.interactive)
            this.mdcRipple = new MDCRipple(this.$el)
    },
    destroyed() {
        this.mdcRipple.destroy()
    }
}
</script>

<style lang="scss">
@import "@material/button/mdc-button";
</style>
