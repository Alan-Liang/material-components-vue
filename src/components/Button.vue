<template>
    <button class="mdc-button" :class="classes" :id="id" v-bind="$attrs" data-mdc-auto-init="MDCRipple">
        {{text}}
    </button>
</template>

<script>
import { MDCRipple } from '@material/ripple'
export default {
    props: {
        text: {
            type: String,
            required: false
        },
        id: {
            type: String,
            required: false
        },
        raised: {
            type: Boolean,
            required: false
        },
        unelevated: {
            type: Boolean,
            required: false
        },
        stroked: {
            type: Boolean,
            required: false
        },
        dense: {
            type: Boolean,
            required: false
        },
        compact: {
            type: Boolean,
            required: false
        },
        primary: {
            type: Boolean,
            required: false
        },
        accent: {
            type: Boolean,
            required: false
        },
    },
    computed: {
        classes() {
            return {
                'mdc-button--raised': this.raised,
                'mdc-button--unelevated': this.unelevated,
                'mdc-button--stroked': this.stroked,
                'mdc-button--dense': this.dense,
                'mdc-button--compact': this.compact,
                'mdc-button--primary': this.primary,
                'mdc-button--accent': this.accent
            }
        }
    },
    mounted() {
        MDCRipple.attachTo(this.$el)
    }
}
</script>

<style lang="scss">
@import "@material/button/mdc-button";
</style>
