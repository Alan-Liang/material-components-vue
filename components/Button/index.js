import Button from './Button.vue'
import './styles.scss'

import { initPlugin } from '../'

const plugin = {
  install (vm) {
    vm.component('m-button', Button)
  }
}
export default plugin

initPlugin(plugin)
