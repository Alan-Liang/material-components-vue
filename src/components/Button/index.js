import Button from './Button'

export { Button }

export default {
  install (vm) {
    vm.component('m-button', Button)
  }
}
