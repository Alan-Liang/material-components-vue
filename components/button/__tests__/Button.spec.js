import 'mutationobserver-shim'
import { mount } from '@vue/test-utils'
import Button from '../Button.vue'

describe('Button', () => {
  it('should mount', () => {
    let wrapper = mount(Button)
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.vm.$data.mdcRipple).toBeDefined()
    expect(wrapper.vm.$data.slotObserver).toBeDefined()
  })

  it('should render with no prop', () => {
    let wrapper = mount(Button)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.classes()).toContain('mdc-button')
  })

  it('should render as raised', () => {
    let wrapper = mount(Button, {
      propsData: {
        raised: true
      }
    })
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.classes()).toContain('mdc-button--raised')
  })

  it('should render as unelevated', () => {
    let wrapper = mount(Button, {
      propsData: {
        unelevated: true
      }
    })
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.classes()).toContain('mdc-button--unelevated')
  })

  it('should render as outlined', () => {
    let wrapper = mount(Button, {
      propsData: {
        outlined: true
      }
    })
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.classes()).toContain('mdc-button--outlined')
  })

  it('should render as dense', () => {
    let wrapper = mount(Button, {
      propsData: {
        dense: true
      }
    })
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.classes()).toContain('mdc-button--dense')
  })
})
