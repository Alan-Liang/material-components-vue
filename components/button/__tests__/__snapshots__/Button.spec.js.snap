// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render as dense 1`] = `<button class="mdc-button mdc-button--dense"> </button>`;

exports[`Button should render as outlined 1`] = `<button class="mdc-button mdc-button--outlined"> </button>`;

exports[`Button should render as raised 1`] = `<button class="mdc-button mdc-button--raised"> </button>`;

exports[`Button should render as unelevated 1`] = `<button class="mdc-button mdc-button--unelevated"> </button>`;

exports[`Button should render with no prop 1`] = `<button class="mdc-button"> </button>`;
